src/ai/backend/manager/services/session/service.py

```diff
@@ -479,22 +475,24 @@ async def _commit_and_upload(reporter: ProgressReporter) -> None:
                             case BgtaskDoneEvent():
                                 break
                             case BgtaskFailedEvent():
-                                raise BackendError(extra_msg=event.message)
+                                return DispatchResult.error(str(event.message))
                             case BgtaskCancelledEvent():
-                                raise BackendError(extra_msg="Operation cancelled")
+                                return DispatchResult.error("Operation cancelled")
 
                 await reporter.update(increment=1, message="Pushed image to registry")
                 # rescan updated image only
-                await rescan_images(
+                rescan_result = await rescan_images(
                     self._db,
                     new_image_ref.canonical,
                     registry_project,
                     reporter=reporter,
                 )
                 await reporter.update(increment=1, message="Completed")
-            except BackendError:
+                rescanned_image_ids = [image.id for image in rescan_result.images]
+                return DispatchResult.success(rescanned_image_ids)
+            except BackendError as e:
                 log.exception("CONVERT_SESSION_TO_IMAGE: exception")
-                raise
+                return DispatchResult.error(str(e))
```

Comment:
It doesn't seem like a good idea that we should always match raising an error here, but we'll get to that later.